src/App/store/AppState.ts

```diff
@@ -194,6 +194,30 @@ export class Project implements BKProject {
     );
   }
 
+  @computed({ keepAlive: true })
+  get isSelected(): boolean {
+    return this.books.some((b: Book) => {
+      return _.some(b.chapters, 'isSelected');
+    });
+  }
+
+  @computed({ keepAlive: true })
+  get allSelected(): boolean {
+    return this.books.every((b: Book) => {
+      return _.every(b.chapters, 'isSelected');
+    });
+  }
+
+  @action.bound
+  toggleAllChapters(): void {
+    const isSelected = this.isSelected;
+    this.books.forEach((b: Book) => {
+      b.chapters.forEach((c: Chapter) => {
+        c.isSelected = !isSelected;
```

Comment:
Is there some reason this doesn't call `c. toggleAllChapters()`?

Reply:
Ok, my first thought when I saw this was that using toggleAllChapters would have been a much better way. But it turns out that the keyword is "toggle". It toggles all the chapters within each book. So if I have some books selected and some not selected then pushing the clear all button will make the unselected ones selected and clear the selected ones. "Toggling" them all.

But I did clean up a couple of the methods above by reusing the book methods.